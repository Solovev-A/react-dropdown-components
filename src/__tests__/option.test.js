import React from 'react';
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom';
import 'jest-styled-components';
import userEvent from "@testing-library/user-event";

import Option from './../lib/components/option';

describe('Option component', () => {
    describe('renders correctly', () => {
        test('without props passed', () => {
            render(<Option />);
            expect(screen.getByRole('option')).toMatchSnapshot();
        });

        test('when pointed', () => {
            window.HTMLElement.prototype.scrollIntoView = jest.fn()
            render(<Option isPointerOver={true} />);

            expect(screen.getByRole('option')).toMatchSnapshot();
        });

        test('when selected', () => {
            render(<Option isSelected={true} />);
            expect(screen.getByRole('option')).toMatchSnapshot();
        });
    });

    it('renders children', () => {
        render(<Option><span>Test</span></Option>);
        expect(screen.getByText(/test/i)).toBeInTheDocument();
    });

    describe('behavior', () => {
        it('use callback on click', () => {
            const onClick = jest.fn();
            const { container } = render(<Option onClick={onClick} />);
            userEvent.click(container.firstChild);
            expect(onClick).toHaveBeenCalledTimes(1);
        });

        it('use callback on mouse enter', () => {
            const onMouseEnter = jest.fn();
            render(<Option onMouseEnter={onMouseEnter} />);
            userEvent.hover(screen.getByRole('option'));
            expect(onMouseEnter).toHaveBeenCalledTimes(1);
        });
    });
});