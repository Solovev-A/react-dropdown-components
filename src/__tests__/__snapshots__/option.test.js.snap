// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option component renders correctly when pointed 1`] = `
.c0 {
  padding: .375rem .75rem;
  display: block;
  white-space: nowrap;
  cursor: pointer;
  color: #fff;
  background-color: #007bff;
}

<span
  class="c0"
  role="option"
/>
`;

exports[`Option component renders correctly when selected 1`] = `
.c0 {
  padding: .375rem .75rem;
  display: block;
  white-space: nowrap;
  cursor: pointer;
  color: #212529;
  background-color: #f2f2f2;
}

<span
  aria-selected="true"
  class="c0"
  role="option"
/>
`;

exports[`Option component renders correctly without props passed 1`] = `
.c0 {
  padding: .375rem .75rem;
  display: block;
  white-space: nowrap;
  cursor: pointer;
}

<span
  class="c0"
  role="option"
/>
`;
